### PR TITLE
add popularity score as Table attribute

### PR DESCRIPTION
**Pull request by dwarszawski**

### CODE REVIEW

File: amundsenatlastypes/schema/01_1_column_schema.json [hunk 1]
```diff
@@ -1,9 +1,59 @@
...
+          "name": "stat_name",
+          "typeName": "string",
+          "isOptional": false,
+          "cardinality": "SINGLE",
+          "isUnique": false,
+          "isIndexable": false
+        },
+        {
+          "name": "stat_val",
+          "typeName": "double",
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_1_column_schema.json]: maybe it's good time to consider having this field in different type than double ? in the future we will have other kinds of statistics (like histograms etc). II propose having stat_val of `string` type and maybe additional field stat_kind which is optional but can contain info about type of stat (like double, string, histogram)

File: amundsenatlastypes/schema/06_column_partition_schema.json [hunk 2]
```diff
@@ -2,28 +2,28 @@
   "entityDefs": [
```
Comment (verdan) [inline amundsenatlastypes/schema/06_column_partition_schema.json]: @dwarszawski @morpheyesh this is what I was talking about. 
Do we need to have a relationship between hive_column and hive_column_partition ? I don't think so. The only relation needed would be hive_colum_partition with hive_table_partition. 
So we can simply add "hive_column" as the superType, and it will all work just fine. No relation needed. 
"hive_column" is already inherited from "Column" which has the statistics and other required changes in place. 


P.S: And also, this is only hive_column_partition, so maybe change the name of the file.

Comment (morpheyesh) [reply amundsenatlastypes/schema/06_column_partition_schema.json]: I do see your point here. This makes sense for table to have such a relationship and maintain table partitions, but does not for columns.

File: amundsenatlastypes/schema/01_1_column_schema.json (hunk 1 above)
Comment (dwarszawski) [reply amundsenatlastypes/schema/01_1_column_schema.json]: out of the scope of this change

File: amundsenatlastypes/schema/06_column_partition_schema.json (hunk 2 above)
Comment (dwarszawski) [reply amundsenatlastypes/schema/06_column_partition_schema.json]: fixed

File: amundsenatlastypes/schema/05_table_partition_schema.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     {
       "name": "TablePartition",
       "superTypes": [
-        "DataSet"
+        "Table"
```
Comment (verdan) [inline amundsenatlastypes/schema/05_table_partition_schema.json]: @dwarszawski I am missing a relationship between Table and TablePartition. 
Under the hive_table, how you'll link the hive_table_partitions ?

Comment (dwarszawski) [reply amundsenatlastypes/schema/05_table_partition_schema.json]: move relation from hive_table <-> hive_table_partition to Table <-> TablePartition

File: amundsenatlastypes/schema/01_column_schema.json [hunk 4]
```diff
@@ -40,21 +40,11 @@
   ],
   "entityDefs": [
     {
+      "name": "Column",
       "superTypes": [
-        "Referenceable"
+        "DataSet"
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_column_schema.json]: why did you change the supertype of Column ? Referencable seemed like appropriate choice

File: amundsenatlastypes/schema/05_table_partition_schema.json [hunk 5]
```diff
@@ -1,29 +1,27 @@
 {
   "entityDefs": [
     {
+      "name": "TablePartition",
       "superTypes": [
-        "TablePartition"
-      ],
-      "name": "hive_table_partition",
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/05_table_partition_schema.json]: why are we removing hive_table_partition ? I think that different systems might also have partitioned tables so we should distinguish those the same way we have hive_table and hbase_table deriving from Table

File: amundsenatlastypes/schema/01_column_schema.json (hunk 4 above)
Comment (dwarszawski) [reply amundsenatlastypes/schema/01_column_schema.json]: it used to be DataSet. The confusion is introduced by the diff itself:
```
{
  "entityDefs": [
    {
      "name": "Column",
      "superTypes": [
        "DataSet"
      ]
    }
  ]
}
```

File: amundsenatlastypes/schema/01_2_table_schema.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        "DataSet"
+      ],
+      "attributeDefs": [
+        {
+          "name": "popularityScore",
+          "typeName": "float",
+          "isOptional": true,
+          "cardinality": "SINGLE",
+          "isUnique": false,
+          "isIndexable": true,
```
[the remark this answers is not in the document]
Comment (dwarszawski) [reply amundsenatlastypes/schema/01_2_table_schema.json]: need to fixed that

File: amundsenatlastypes/schema/05_table_partition_schema.json (hunk 5 above)
Comment (dwarszawski) [reply amundsenatlastypes/schema/05_table_partition_schema.json]: again - it is still there but in different file.

File: amundsenatlastypes/schema/06_column_partition_schema.json (hunk 2 above)
Comment (verdan) [reply amundsenatlastypes/schema/06_column_partition_schema.json]: Do we even need this hive_column_partition ?

Comment (dwarszawski) [reply amundsenatlastypes/schema/06_column_partition_schema.json]: That would be useful in the future but now is useless. Removed!
